Added output file controls and sanitization

inspectorG4dget opened the pull request:
Added the '-o' flag to specify the output filepath
Added filename sanitization to replace illegal characters in the filename

inspectorG4dget committed:
Added output file controls and sanitization

Added the '-o' flag to specify the output filepath
Added filename sanitization to replace illegal characters in the filename

ids1024 commented on scripts/ytdl:
```diff
@@ -87,7 +88,18 @@ def download(video, audio=False, stream=None, preftype=None):
+
+    outdirpath = os.path.dirname(outfilepath)
+    outfilename = os.path.basename(outfilepath)
+    validChars = set('-_. ' + string.ascii_letters)
```
I think this is too restricted; non-acsii characters can be valid filenames (imagine downloading something with a name in Chinese/Russian/etc.). And this eliminates even numbers.

inspectorG4dget committed:
Removed outfilepath validation as it was too restrictive

Previously, the output filepath was restricted to only a subset of ascii characters. That restriction has now been lifted, but the ability to specify an output file path with the "-o" flag still remains

inspectorG4dget committed:
Code Cleanup

Removed a deprecated import and fixed a minor typo-bug

inspectorG4dget commented:
I've removed the excessive filename sanitization, leaving in the useful features (ability to specify output filepath)
